core/cmake/BareosGenerateDebianInfo.cmake

```diff
@@ -17,98 +17,161 @@
+       DEBIAN_CONTROL_BAREOS_FILEDAEMON
+  )
+  file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-universal-client-dbg
+       DEBIAN_CONTROL_BAREOS_DBG
+  )
+else()
 
-  if(HAVE_GLUSTERFS)
-    file(READ
-         ${CMAKE_SOURCE_DIR}/debian/control.bareos-filedaemon-glusterfs-plugin
-         DEBIAN_CONTROL_FILEDAEMON_GLUSTERFS_PLUGIN
+  if(NOT client-only)
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos DEBIAN_CONTROL_BAREOS)
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-database
+         DEBIAN_CONTROL_BAREOS_DATABASE
     )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-glusterfs
-         DEBIAN_CONTROL_STORAGE_GLUSTERFS
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-director
+         DEBIAN_CONTROL_BAREOS_DIRECTOR
     )
-  endif()
-
-  if(HAVE_CEPHFS)
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-filedaemon-ceph-plugin
-         DEBIAN_CONTROL_FILEDAEMON_CEPH_PLUGIN
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage
+         DEBIAN_CONTROL_BAREOS_STORAGE
     )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-ceph
-         DEBIAN_CONTROL_STORAGE_CEPH
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-tools
+         DEBIAN_CONTROL_BAREOS_TOOLS
     )
   endif()
 
-  if(TARGET droplet)
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-droplet
-         DEBIAN_CONTROL_STORAGE_DROPLET
-    )
-  endif()
+  file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-filedaemon
+       DEBIAN_CONTROL_BAREOS_FILEDAEMON
+  )
+  file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-common
+       DEBIAN_CONTROL_BAREOS_COMMON
+  )
+  file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-dbg
+       DEBIAN_CONTROL_BAREOS_DBG
+  )
 
-  # python (2) plugins
-  if(Python2_FOUND)
-    file(READ
-         ${CMAKE_SOURCE_DIR}/debian/control.bareos-filedaemon-python2-plugin
-         DEBIAN_CONTROL_FILEDAEMON_PYTHON_PLUGIN
+  if(ENABLE_BCONSOLE)
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-client
+         DEBIAN_CONTROL_BAREOS_CLIENT
     )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-python2-plugin
-         DEBIAN_CONTROL_STORAGE_PYTHON_PLUGIN
-    )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-director-python2-plugin
-         DEBIAN_CONTROL_DIRECTOR_PYTHON_PLUGIN
+    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-bconsole
+         DEBIAN_CONTROL_BAREOS_BCONSOLE
     )
   endif()
 
-  # python 3 plugins
-  if(Python3_FOUND)
-    file(READ
-         ${CMAKE_SOURCE_DIR}/debian/control.bareos-filedaemon-python3-plugin
-         DEBIAN_CONTROL_FILEDAEMON_PYTHON3_PLUGIN
-    )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-storage-python3-plugin
-         DEBIAN_CONTROL_STORAGE_PYTHON3_PLUGIN
-    )
-    file(READ ${CMAKE_SOURCE_DIR}/debian/control.bareos-director-python3-plugin
-         DEBIAN_CONTROL_DIRECTOR_PYTHON3_PLUGIN
-    )
-  endif()
+  if(GENERATE_DEBIAN_CONTROL)
```

Comment:
Is this change of logic intentional ?

Reply:
I completely changed the creation of `debian/control`.
I think the new variant is easier to understand - what do you think?

Reply:
Yes.